frontend/ui/elements/screensaver_menu.lua

```diff
@@ -26,114 +26,144 @@ local function genMenuItem(text, setting, value, enabled_func, separator)
-            return G_reader_settings:isTrue("screensaver_show_message")
-        end,
-        callback = function()
-            G_reader_settings:toggle("screensaver_show_message")
-        end,
-        separator = true,
-    },
-    -- separator
     {
-        text = _("Settings"),
+        text = _("Screensaver"),
         sub_item_table = {
+            genMenuItem(_("Show book cover on sleep screen"), "screensaver_type", "cover", hasLastFile),
+            genMenuItem(_("Show custom image on sleep screen"), "screensaver_type", "image_file"),
+            genMenuItem(_("Shuffle images on sleep screen"), "screensaver_type", "random_image"),
+            genMenuItem(_("Show document cover on sleep screen"), "screensaver_type", "document_cover"),
+            genMenuItem(_("Show reading progress on sleep screen"), "screensaver_type", "readingprogress", isReaderProgressEnabled),
+            genMenuItem(_("Show book status on sleep screen"), "screensaver_type", "bookstatus", hasLastFile),
+            genMenuItem(_("Lock the screen in current state"), "screensaver_type", "disable", nil, true),
+            separator = true,
             {
-                text = _("Screensaver folder"),
-                keep_menu_open = true,
-                callback = function()
-                    Screensaver:chooseFolder()
-                end,
+                text = _("Border fill"),
+                sub_item_table = {
+                    genMenuItem(_("Black"), "screensaver_img_background", "black"),
+                    genMenuItem(_("White"), "screensaver_img_background", "white"),
+                    genMenuItem(_("Current state"), "screensaver_img_background", "none", nil, true),
+                    -- separator
+                        {
+                            text_func = function()
+                                local percentage = G_reader_settings:readSetting("screensaver_stretch_limit_percentage")
+                                if G_reader_settings:isTrue("screensaver_stretch_images") and percentage then
+                                    return T(_("Stretch to fit screen (with limit: %1 %)"), percentage)
+                                end
+                                return _("Stretch cover to fit screen")
+                            end,
+                            checked_func = function()
+                                return G_reader_settings:isTrue("screensaver_stretch_images")
+                            end,
+                            callback = function(touchmenu_instance)
+                                Screensaver:setStretchLimit(touchmenu_instance)
+                            end,
+                        },
+                },
             },
             {
-                text = _("Screensaver image"),
-                keep_menu_open = true,
-                callback = function()
-                    Screensaver:chooseFile()
-                end,
+                text = _("Unlock screen delay"),
```

Comment:
Some "lock" notion here - although it feels more ok reading "unlock" than "lock screen".
`Exit sleep screen delay` ?

Reply:
I don't really like `wake up` but unlock does not seem harmful to me either.

Reply:
What's this delay?

Reply:
it delays the normal UI from coming to life and keeps the sleep screen on after you press the power button.

Reply:
Okay, so the current text is the self-explanatory if somewhat awkward `Keep the screensaver on screen after wakeup`, introduced in <https://github.com/koreader/koreader/pull/7334>. (This is the kind of thing that would be clear with a constrained scope. ^_^)

I think the fact that it's about the delay coming from sleep is vital information here.

E.g., something like `Maintain sleep screen after wake up`.